src/builderGenerator.ts

```diff
@@ -262,7 +262,9 @@ function addBuilderMethodToImports({
 }): { name: string; statement: string } {
   if (
     !isUnion &&
-    type.isClassOrInterface() &&
+    !type.isArray() &&
+    !isPrimitiveType(type) &&
+    !type.getText().startsWith('{') &&
```

Comment:
I used it to handle the case with anonymous nested object like
```
{
  a: {
    b: boolean
  }
}
```

this one was broken because when type name is queried it returned in `null` this breaking the generated output

so thats not a check for a union but rather for anything but named nested object-like type
maybe it worth wrapping in separate method...

wdyt?

Reply:
`type a = {a: string} | a2`